homeassistant/components/kostal_piko/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""Kostal Piko custom component."""
+from datetime import timedelta
+import logging
+from math import ceil
+
+from kostal import InfoVersions, Piko, SettingsGeneral
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["sensor"]
+DEVICE_INFO_IDS = [
+    SettingsGeneral.INVERTER_NAME,
+    SettingsGeneral.INVERTER_MAKE,
+    InfoVersions.SERIAL_NUMBER,
+    InfoVersions.ARTICLE_NUMBER,
+    InfoVersions.VERSION_FW,
+    InfoVersions.VERSION_HW,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Kostal Piko from a config entry."""
+    host = entry.data[CONF_HOST]
+    user = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+
+    try:
+        piko = Piko(async_get_clientsession(hass), host, user, password)
+    except Exception as err:
```

Comment:
Can you seperate invalid auth and other exceptions? It is a lot more helpful for the user that way

Reply:
This is done in the config_flow (`async_step_user`), where the connection is tested. When the Piko is first created, there is no actual request sent. So there is also no way of knowing any auth issues, since its active polling.